src/components/profile-view/profile-view.jsx

```diff
@@ -0,0 +1,257 @@
+                localStorage.setItem('user', data.Username);
+                // Reload the page to make sure that the user can immediately start using their new details
+                window.open(`/users/${data.Username}`, '_self');
+            }).catch(error => {
+                console.log('error updating user details')
+            });
+        }
+    };
+
+    // Function for updating the appropriate state variables as the user fills out the update form
+    handleFieldChange(event) {
+        let {name, value} = event.target;
+        this.setState({ [name]: value})
+    }
+
+    // Function to show the modal that confirms you want to delete a user profile
+    showModal() {
+        this.setState({ modalState: true });
+    }
+
+    // Fuunction for closing the modal that confirms you want to delete a user profile
+    closeModal() {
+        this.setState({ modalState: false });
+    }
+
+    // Function for deleting user details. A delete request is made ot the API for this user
+    deleteUserDetails() {
+        let token = localStorage.getItem('token');
+        let user = localStorage.getItem('user');
+        axios.delete(`https://t-dogg-movies-api.herokuapp.com/users/${user}`, {
+            headers: { Authorization: `Bearer ${token}`}
+        }).then(response => {
+            const data = response.data;
+            alert(user + " has been deleted");
+            // Remove the user details and auth token from localStorage, and send the user back to the login page (since they are now logged out)
+            localStorage.removeItem('user');
+            localStorage.removeItem('token');
+            window.open('/', '_self');
+        }).catch(error => {
+            console.log('error deleting the user');
+        })
+    }
+
+    // Render function to display items on the DOM
+    render() {
+        // Get the props that were passed into this view and store them in appropriate variables
+        const { movies, onBackClick} = this.props;
+
+        // Section of code for getting the users favourites (so that they can be displayed on the page)
+        // I am aware that this probably isn't the best way/place to get this information, but I couldn't work out another way to do it
+        // In hindsight, if I was writing this again I'd use a function component for ProfileView instead of a class component to make working with this information easier
+        // First get the array of a user's favourite movies (which was obtained from the initial GET request to the API)
+        let tempArray = this.state.favouriteMovies;
+        // Get an empty array which will store all of the movie objects which match the favourites list
+        let favouriteMoviesArray = [];
+        // Filter the movies array (obtained from props) and only save those movies which match ID's from the list of the users favourites
+        favouriteMoviesArray = movies.filter(movie => tempArray.includes(movie._id));
+
```

Comment:
The above logic should not be in the render function. Instead write a class function and trigger it in lifecycle methods for mount and/or update (componentDidMount / componentDidUpdate)